File: Day20_Beauty_Sum_of_a_String/beauty_sum.py
class Solution(object):
    def beautySum(self, s):
        """
        :type s: str
        :rtype: int
        """

        ans=0
        for i in range(len(s)):
            freq_d={}
            for j in range(i,len(s)):
                if s[j] in freq_d:
                    freq_d[s[j]]+=1
                else:
                    freq_d[s[j]]=1
                ans+=(max(freq_d.values())-min(freq_d.values()))    
        return ans
        

File: Day20_Beauty_Sum_of_a_String/beauty_sum.java
class Solution {
    public int beautySum(String s) {
        int ans=0;
        for(int i=0;i<s.length();i++){
            HashMap<Character,Integer> m=new HashMap<>();
            for(int j=i;j<s.length();j++){
                if(m.containsKey(s.charAt(j))){
                    m.put(s.charAt(j),m.get(s.charAt(j))+1);
                }
                else{
                    m.put(s.charAt(j),1);
                }
                int mini=Collections.min(m.values());
                int max=Collections.max(m.values());
                ans+=(max-mini);
            }
        }
        return ans;
    }
}
